xla/tests/onednn_matmul_test.cc

```diff
@@ -520,6 +529,31 @@ TEST_F(MatmulTest, TestF32ConstantWeights) {
   )");
 }
 
+TEST_F(MatmulTest, SimpleTestF32WithMulAndAddFusion) {
+  const char* matmul_module_str = R"(
+  HloModule matmul.mul.add.test.f32, entry_computation_layout={(f32[32,32,40,30]{3,2,1,0}, f32[32,32,30,40]{3,2,1,0})->f32[32,32,40,40]{3,2,1,0}}
+  ENTRY matmul.mul.add.test.f32 {
+    arg0.1 = f32[32,32,40,30]{3,2,1,0} parameter(0), parameter_replication={false}
+    arg0.2 = f32[32,32,30,40]{3,2,1,0} parameter(1), parameter_replication={false}
+    dot.7 = f32[32,32,40,40]{3,2,1,0} dot(arg0.1, arg0.2), lhs_batch_dims={0,1}, lhs_contracting_dims={3}, rhs_batch_dims={0,1}, rhs_contracting_dims={2}
+    const.0 = f32[] constant(0.044715)
+    bcast.0 = f32[32,32,40,40]{3,2,1,0} broadcast(const.0), dimensions={}
+    mul.0 = f32[32,32,40,40]{3,2,1,0} multiply(dot.7,bcast.0)
+    const.1 = f32[] constant(0.65)
+    bcast.1 = f32[32,32,40,40]{3,2,1,0} broadcast(const.1), dimensions={}
+    add.0 = f32[32,32,40,40]{3,2,1,0} add(mul.0, bcast.1)
+    const.2 = f32[] constant(0.65)
+    bcast.2 = f32[32,32,40,40]{3,2,1,0} broadcast(const.2), dimensions={}
+    add.1 = f32[32,32,40,40]{3,2,1,0} add(bcast.2, bcast.1)
+    tuple.12 = (f32[32,32,40,40]{3,2,1,0}) tuple(add.0)
+    tuple.14 = (f32[32,32,40,40]{3,2,1,0}) tuple(add.1)
```

Comment:
Does this unused tuple and the unary tuple-gte serve any purpose? If not, let's remove them.